src/hooks/useScrollShadow.js

```diff
@@ -0,0 +1,94 @@
+// very difficult to test with JSDom, some basic interaction is tested in ScrollableContainer
+/* istanbul ignore file */
+import { useEffect } from 'react'
+import throttle from 'lodash.throttle'
+
+const BOX_SHADOW_INITIAL = '0 0 0 1px rgba(0, 0, 0, 0.1)'
+const BOX_SHADOW_SCROLLED =
+  '0 0 0 1px rgba(0, 0, 0, 0.1), 0 0 0 5px rgba(0, 0, 0, 0.05)'
+
+export default function useScrollShadow({
+  bottomRef,
+  scrollableRef,
+  topRef,
+  shadows = {},
+}) {
+  const initialShadow = shadows.initial || BOX_SHADOW_INITIAL
+  const scrolledShadow = shadows.scrolled || BOX_SHADOW_SCROLLED
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (scrollableRef && scrollableRef.current != null) {
+        const topElement = topRef && topRef.current
+        const bottomElement = bottomRef && bottomRef.current
+
+        if (topElement) {
+          topElement.style.boxShadow = 'var(--scroll-shadow)'
+        }
+
+        if (bottomElement) {
+          bottomElement.style.boxShadow = 'var(--scroll-shadow)'
+        }
+
+        const { isBottomScrolled } = handleShadows(scrollableRef)
+
+        if (isBottomScrolled) {
+          bottomElement.style.setProperty('--scroll-shadow', scrolledShadow)
+        }
+      }
+    }, 100)
+  }, [scrollableRef, bottomRef, topRef, scrolledShadow])
+
+  const handleOnScroll = throttle(e => {
+    const { isTopScrolled, isBottomScrolled } = handleShadows(scrollableRef)
+    const topElement = topRef && topRef.current
+    const bottomElement = bottomRef && bottomRef.current
+
+    if (topElement) {
```

Comment:
this if and the one below could be extracted to a function to avoid some duplication (just passing element, or even ref, and `isScrolled` flag as parameters)

Reply:
Thanks 👌